news/_posts/2020-02-04-the-new-prestashop-test-framework.md

```diff
@@ -0,0 +1,96 @@
+Note: this article is focused on the Core only, but that’s just a small part of what the QA team is working on. More to come in a future article !
+
+
+## What was before
+
+Historically, the PrestaShop Core continuous integration had a big end-to-end test campaign which purpose was to test the whole application as thoroughly as possible. This campaign was using Selenium as the automation tool, and we were running it in two places:
+
+- The whole campaign was running on every nightly build and for every release
+- A small subset of the campaign was running for every Pull Request on GitHub through Travis CI
+
+At the beginning of 2019, the QA team decided to drop this campaign and start over. In September 2019, we started the new version.
+
+Why? There were mainly 3 reasons:
+
+- Tests are run in headless mode (mandatory on Travis) and sadly, Selenium is kind of picky in headless. There were a lot of problems with it, and as you can imagine, it’s way more complicated to debug a failing test when there’s no GUI ! Screenshots have been tried, logs, everything possible but ultimately the failing tests had to be disabled as it was impossible to find a good solution.
+- Selenium itself is not very stable: the test campaign can be run 3 or 4 times in a row on Travis and have different results, with different tests failing. Sometimes it couldn’t find a selector, or timed out when waiting for something… it was very difficult and time-consuming to make the tests more robust.
+- The campaign was too complex: it was trying to test everything at once. The campaign would test all the functionalities in a page, but also doing some [functional testing](https://devdocs.prestashop.com/1.7/testing/how-to-create-your-own-web-acceptance-tests/#functional-tests) (verifying some functionalities work well together) and some [end-to-end testing](https://devdocs.prestashop.com/1.7/testing/how-to-create-your-own-web-acceptance-tests/#end-to-end-tests-soon) (running through the application like a real user and testing a bunch of functionalities at once). The result was a very complex test suite, very brittle (end-to-end tests are by design really fragile) and a nightmare to maintain and fix.
+
+There were some concerns with the code too: although it was working, there was no real framework behind it. No separation of concerns, no proper use of inheritance, no distinction between the test logic and the page logic. It was very complicated to find the source of a problem sometimes, and with more than 8000 test cases it frequently was not possible to fix it.
+
+All these reasons made it clear that it was time to drop the whole thing and start over.
+
+
+## What is coming
+
+The QA team decided to work on three points.
```

Comment:
```suggestion
The QA team decided to work on the three following points.
```

Reply:
As the list is not numbered it makes it more obvious that it is detailed in the following paragpraghs